pkgs/by-name/mi/miru/darwin.nix

```
@@ -0,0 +1,27 @@
+{
+  stdenvNoCC,
+  fetchurl,
+  unzip,
+
+  pname,
+  version,
+  meta,
+}:
+stdenvNoCC.mkDerivation rec {
+  inherit pname version meta;
+
+  src = fetchurl {
+    url = "https://github.com/ThaUnknown/miru/releases/download/v${version}/mac-Miru-${version}-mac.zip";
+    hash = "sha256-OakGB5Fz1Tlxa/Uu7xHlKoEF9VRfWFQ9CjsR0eCRyQw=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,Applications}
+    cp -r Miru.app $out/Applications/
+    ln -s $out/Applications/Miru.app/Contents/MacOS/Miru $out/bin/miru
+  '';
```

```suggestion
  nativeBuildInputs = [
    unzip
    makeWrapper
  ];

  installPhase = ''
    runHook preInstall

    mkdir -p $out/{bin,Applications}
    cp -r Miru.app $out/Applications/
    makeWrapper $out/Applications/Miru.app/Contents/MacOS/Miru $out/bin/miru

    runHook postInstall
  '';
```

`$out/bin/miru` won't work as expected when created with `ln -s`.